### PR TITLE
2002 uta bugfixes

### PR DESCRIPTION
**Pull request by jaybouche**
Fixed profile - car 
by adding fields in the UI to support creating a car;
ability to create or update car upon submitting data;
Fixed profile - membership 
by adding function to button
(requires the back-end getter / setters corrected in User model from isDriver to IsDriver to be fully functional).
Existing comments/ logging/ deleting of non-functional code addressed in documentation-logging branch.

### CODE REVIEW

**Review by vyklimavicius — APPROVED** [review]
Looks good to me

**Review by ingarcia1 — APPROVED** [review]
Profile-car is now functional

**Review by zayn37 — APPROVED** [review]

**Review by zayn37 — COMMENTED** [review]
looks good

**Review by faysalanis97 — APPROVED** [review]
Everything looks clean and seems to be working

**Review by sfjsss — CHANGES_REQUESTED** [review]
line 79 in car.service.ts would try to redirect page to currentUrl+/car which is not available and hence will redirect me back to home page. remove that line would fix the issue.

I don't quite understand the membership backend part, what do you mean by changing from isDriver to IsDriver?

**Comment by jaybouche** [comment]
> line 79 in car.service.ts would try to redirect page to currentUrl+/car which is not available and hence will redirect me back to home page. remove that line would fix the issue.
> 
> I don't quite understand the membership backend part, what do you mean by changing from isDriver to IsDriver?

It is how Jackson looks at the JSON.

**Comment by jaybouche** [comment]
> line 79 in car.service.ts would try to redirect page to currentUrl+/car which is not available and hence will redirect me back to home page. remove that line would fix the issue.
Routing has been removed

**Review by JosueDOrtiz — APPROVED** [review]
Good job. One of my concerns would be the console.log statements that were left in. Just don't forget to clean some of these up once y'all are done working on these bugs. Also, I noticed that Address 2 in the profile-form is retrieving address 2 from work address (wAddress). This is not something y'all changed and there is no adressTwo entity in the user model. Might be something that comes up as y'all keep fixing bugs.

**Review by sfjsss — APPROVED** [review]
looks good now

**Comment by jaybouche** [comment]
Working the bug for location and will proceed after considering needing waddress information. 
Console.log will be removed throughout code upon completion of Sprint; will leave till for debugging purposes at this time.

**Comment by sfjsss** [comment]
@jaybouche 
I personally strongly recommend disregard wAddress or leave it as it is. Enabling this "work address" along with "home address" creates a lot of confusion for users at the same time it doesn't really make any sense within our context considering a lot of associates' home is in a completely different state. In addition, enabling this "work address" will create extra complexity and require a lot of refactoring of our code and also the legacy code at the same time again it doesn't necessarily provide any improvement on user experience or functionality.

**Comment by JosueDOrtiz** [comment]
image
@sfjsss - the address pulled for Address 2 in the JSON is wAddress

**Comment by sfjsss** [comment]
yea. For my validation, I'm also using the wAddress here as "address 2". That's why I suggest leave all the Ws as it is right now otherwise there are potentially a lot of refactoring on both our code and the legacy code.

File: src/app/components/profile-location/profile-location.component.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export class ProfileLocationComponent implements OnInit {
   zipcode: number;
   city:string;
   address:string;
-  address2:string;
+  address2:any;
```
Comment (carolynrehm) [inline src/app/components/profile-location/profile-location.component.ts]: why was this type changed?

**Comment by carolynrehm** [comment]
> Working the bug for location and will proceed after considering needing waddress information.
> Console.log will be removed throughout code upon completion of Sprint; will leave till for debugging purposes at this time.

console logs for you guys to debug is fin, but they shouldn't make it into the dev branch

File: src/app/components/profile-location/profile-location.component.ts (hunk 1 above)
Comment (jaybouche) [reply src/app/components/profile-location/profile-location.component.ts]: Will change to reflect string

**Pull request merged**